### ISSUE TITLE
Support for wsl2 / tar.gz files for Windows platform.

### ISSUE THREAD

**Issue by cdrage** [osbuild/bootc-image-builder#172]
We have:

* Mac = applehv / raw
* Linux = qcow2 / raw

But nothing native for Windows.

Would be awesome to be able to generate wsl2 / .tar.gz images for native Windows support.

**Comment by veillard** [comment]
Yes I think that's a very common use case and possibly the largest population we can target from Podman Desktop so it would absolutely make sense, Plus it's a common use case that even customers have raised.

**Comment by veillard** [comment]
I think there is 2 required sub issues:
1/ Support for WSL2 in the image builder container image: this is already a preview in the online version so hopefully this is an easy step
2/ Support launching WSL2 'images' (a tarball mostly) from the lugin like we do for ISO or raw on Windows platform

**Comment by deboer-tim** [comment]
Pinging on this again. The Macadam CLI and Podman Desktop bootc extension now have the ability to create and run VMs from disk images, so if bib generated WSL2 we'd have an end-to-end flow from containerfile to running VM on Windows for the first time.

I know Windows isn't exactly the target market for Linux VMs :), but it's still the OS used by >50% of enterprise developers. So, this would double the number of developers who can easily try out bootc and confirm it works.

**Comment by cdrage** [comment]
@ondrejbudai @veillard @deboer-tim 

Did some investigation and I was able to get a bootc image booting by doing the following:

1. Used `podman export -o foobar.wsl quay.io/fedora/fedora-bootc:42`
2. Modified the resulting WSL file by adding the configuration file by following: https://learn.microsoft.com/en-us/windows/wsl/wsl-config
2. Successfully booted it on Windows

**HOWEVER**. I was unable to login / ssh / etc into it, as we are lacking support for the build config YAML file.

The build config yaml is required in order for us to setup username/password/ssh in the resulting image file (iso, raw, qcow2, etc.). 

I'm assuming we'd have to add this to `bootc-image-builder` as the build config builds / modifies the image according to the build config.
